nvtabular/tools/dataset_inspector.py

```diff
@@ -0,0 +1,128 @@
+        else:
+            return super(NpEncoder, self).default(obj)
+
+
+class DatasetInspector:
+    """
+    Analyzes an existing dataset to extract its statistics.
+    """
+
+    def inspect(self, path, dataset_format, columns_dict, devices, output_file):
+        """
+        Parameters
+        -----------
+        path: str, list of str, or <dask.dataframe|cudf|pd>.DataFrame
+            Dataset path (or list of paths), or a DataFrame. If string,
+            should specify a specific file or directory path. If this is a
+            directory path, the directory structure must be flat (nested
+            directories are not yet supported).
+        dataset_format: str
+            Dataset format (i.e parquet or csv)
+        columns_dict: dict
+            Dictionary indicating the diferent columns type
+        output_file: str
+            Filename to write the output statistics
+        """
+        # Get dataset columns
+        cats = columns_dict["cats"]
+        conts = columns_dict["conts"]
+        labels = columns_dict["labels"]
+        all_cols = cats + conts + labels
+
+        # Dictionary with json output key names
+        key_names = {}
+        key_names["min"] = {}
+        key_names["min"]["cats"] = "min_entry_size"
+        key_names["min"]["conts"] = "min_val"
+        key_names["max"] = {}
+        key_names["max"]["cats"] = "max_entry_size"
+        key_names["max"]["conts"] = "max_val"
+        key_names["mean"] = {}
+        key_names["mean"]["cats"] = "avg_entry_size"
+        key_names["mean"]["conts"] = "mean"
+
+        # Create Dataset, Workflow, and get Stats
+        dataset = Dataset(path, engine=dataset_format)
+        features = all_cols >> DataStats()
+        with managed_client(devices) as client:
+            workflow = Workflow(features, client=client)
+            workflow.fit(dataset)
+
+        # Save stats in a file and read them back
+        stats_file = "stats_output.yaml"
+        workflow.save_stats(stats_file)
+        output = yaml.safe_load(open(stats_file))
+        output = output[1]["stats"]
+
+        # Dictionary to store collected information
+        data = {}
+        # Store num_rows
+        data["num_rows"] = dataset.num_rows
+        # Store cols
+        for col_type in ["conts", "cats", "labels"]:
+            data[col_type] = {}
+            for col in columns_dict[col_type]:
+                data[col_type][col] = {}
+                data[col_type][col]["dtype"] = output[col]["dtype"]
+                if col_type != "conts":
+                    data[col_type][col]["cardinality"] = output[col]["cardinality"]
+                if col_type != "labels":
+                    data[col_type][col][key_names["min"][col_type]] = output[col]["min"]
+                    data[col_type][col][key_names["max"][col_type]] = output[col]["max"]
+                    data[col_type][col][key_names["mean"][col_type]] = output[col]["mean"]
+                if col_type == "conts":

```

Comment:
Nitpick: the key_names structure is confusing - and removing it in favour of an if/elif will reduce the number of lines slightly:

```suggestion
                if col_type == "cats":
                    data[col_type][col]["min_entry_size"] = output[col]["min"]
                    data[col_type][col]["max_entry_size"] = output[col]["max"]
                    data[col_type][col]["avg_entry_size"] = output[col]["mean"]
                elif col_type == "conts":
                    data[col_type][col]["min_val"] = output[col]["min"]
                    data[col_type][col]["max_val"] = output[col]["max"]
                    data[col_type][col]["mean"] = output[col]["mean"]
```